python/ambassador/ir/ir.py

```diff
@@ -480,7 +480,11 @@ def __init__(
             if len(name) > 60:
                 # Too long. Gather this cluster by name prefix and normalize
                 # its name below.
-                short_name = name[0:40]
+                h = hashlib.new("sha1")
+                h.update(name.encode("utf-8"))
+                hd = h.hexdigest()[0:16].upper()
```

Comment:
Just trying to make sure I understand this due to my lack of python skills. Envoy limits us to 60 characters so when it is greater than that we need to shorten and merge clusters.

The first 39 characters + 1 static dash ("-") + 15 character hash so we will shorten it to 55 characters...So did I interpret my python code correctly? If so any reason for only allowing 55 characters rather than full 60?

Reply:
`name[0:40]` is actually the first 40 characters rather than the first 39, so it's 40 + "-" + 16 for the hash + `-0` or 59 characters, which is close enough that I didn't want to push my luck. 😉 

(The `-0` is all but certainly unneeded at this point, too, but I still need to finish convincing myself of that.)

Reply:
Ok, that makes sense and understood. Yeah, I would think introducing the hash negates any reason for keeping the suffix around. Do you want to remove that as part of this PR or follow-up when you have had more time to think of the repercussions of that change?

Reply:
The latter, yeah. Hopefully soon. 🤞